pypeit/fluxspec.py

```diff
@@ -555,3 +558,390 @@ def __repr__(self):
+      Used only to set settings for calls to the Class outside of PyepIt
+    sens_file : str
+      Filename of a sensitivity function file to be input
+    setup : str
+      Setup name (for MasterFrame)
+    settings : dict-like
+      Settings to guide the fluxing
+      Key ones are ['mosaic']['longitude', 'latitude', 'elevation']
+
+
+    Attributes
+    ----------
+    sensfunc : dict
+      Sensitivity function
+
+    steps : list
+      List of steps performed
+    frametype : str
+      Set to 'sensfunc'
+    std : SpecObj
+      The chosen one for generating the sensitivity function
+    std_header : dict-like
+      Used for the RA, DEC, AIRMASS, EXPTIME of the standard star spectrum
+    std_idx : int or list
+      Index that std is within the std_specbojs list
+    sci_specobjs : SpecObjs
+      List of SpecObj objects to be fluxed (or that were fluxed)
+    sci_header : dict-like
+      Used for the airmass, exptime of the science spectra
+    spectrograph : Spectrograph
+      Used for extinction correction
+    """
+
+    # Frametype is a class attribute
+    frametype = 'sensfunc'
+
+    def __init__(self, std_spec1d_file=None, sci_spec1d_file=None, sens_file=None,
+                 std_specobjs=None, std_header=None, spectrograph=None,
+                 telluric=False, setup=None, master_dir=None, mode=None,
+                 star_type=None, star_mag=None, BALM_MASK_WID=5.0, nresln=None, debug=False):
+
+        # Load standard files
+        std_spectro = None
+        self.std_spec1d_file = std_spec1d_file
+        # Need to unwrap these (sometimes)..
+        self.std_specobjs = std_specobjs
+        self.std_header = std_header
+        if self.std_spec1d_file is not None:
+            self.std_specobjs, self.std_header = load.ech_load_specobj(self.std_spec1d_file)
+            msgs.info('Loaded {0} spectra from the spec1d standard star file: {1}'.format(
+                len(self.std_specobjs), self.std_spec1d_file))
+            std_spectro = self.std_header['INSTRUME']
+
+        try:
+            self.std_ra = self.std_header['RA']
+        except:
+            self.std_ra = None
+        try:
+            self.std_dec = self.std_header['DEC']
+        except:
+            self.std_dec = None
+        try:
+            self.std_file = self.std_header['FILENAME']
+        except:
+            self.std_file = None
+
+        # Load the science files
+        sci_spectro = None
+        self.sci_spec1d_file = sci_spec1d_file
+        self.sci_specobjs = []
+        self.sci_header = None
+        if self.sci_spec1d_file is not None:
+            self.sci_specobjs, self.sci_header = load.ech_load_specobj(self.sci_spec1d_file)
+            msgs.info('Loaded {0} spectra from the spec1d science file: {1}'.format(
+                len(self.sci_specobjs), self.sci_spec1d_file))
+            sci_spectro = self.sci_header['INSTRUME']
+
+        # Compare instruments if they exist
```

Comment:
I'm lost here, when will we ever be fluxing from a different instrument?

Reply:
we used the coordinates of the telescope to correct the extinction.